coriolis/tests/api/middleware/test_auth.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2022 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+from unittest import mock
+
+from oslo_middleware import request_id
+from oslo_serialization import jsonutils
+
+from coriolis.api.middleware.auth import CoriolisKeystoneContext
+from coriolis.api import wsgi
+from coriolis import context
+from coriolis.tests import test_base
+
+
+class CoriolisKeystoneContextTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis api middleware auth."""
+
+    @mock.patch.object(jsonutils, "loads")
```

Comment:
You don't really need to mock this. This only turns a JSON-formatted string into a `dir`. Instead of mocking its return value, you could do the following:
```python
expected_service_catalog = {"catalog1": ["service1", "service2"]}
req_mock.headers = {
    # ...
    'X_SERVICE_CATALOG': str(expected_service_catalog), # this will turn your dir into a string
    # ...
}
# ...
# ...
mock_request_context.assert_called_once_with(
    # ...
    service_catalog=expected_service_catalog, # checks that the string from headers is the original dir
    # ...
)
```